pkg/cmd/evict_pod.go

```diff
@@ -4,55 +4,57 @@ import (
 	"fmt"
 	"os"
 
-	"github.com/rajatjindal/kubectl-evict-pod/pkg/k8s"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 	"k8s.io/client-go/kubernetes"
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-//Version is set during build time
+// Version is set during build time
 var Version = "unknown"
 
-//EvictPodOptions is struct for modify secret
+// EvictPodOptions is struct for modify secret
 type EvictPodOptions struct {
 	configFlags *genericclioptions.ConfigFlags
-	iostreams   genericclioptions.IOStreams
-
-	args         []string
-	podName      string
-	namespace    string
-	kubeclient   kubernetes.Interface
-	printVersion bool
+	ioStreams   genericclioptions.IOStreams
+
+	podNames      []string
+	namespace     string
+	kubeclient    kubernetes.Interface
+	printVersion  bool
+	labelSelector string
+	fieldSelector string
 }
 
 // NewEvictPodOptions provides an instance of EvictPodOptions with default values
 func NewEvictPodOptions(streams genericclioptions.IOStreams) *EvictPodOptions {
 	return &EvictPodOptions{
 		configFlags: genericclioptions.NewConfigFlags(true),
-		iostreams:   streams,
+		ioStreams:   streams,
 	}
 }
 
-// NewCmdModifySecret provides a cobra command wrapping EvictPodOptions
-func NewCmdModifySecret(streams genericclioptions.IOStreams) *cobra.Command {
+// NewEvictCmd provides a cobra command wrapping EvictPodOptions
+func NewEvictCmd(streams genericclioptions.IOStreams) *cobra.Command {
 	o := NewEvictPodOptions(streams)
 
 	cmd := &cobra.Command{
 		Use:          "evict-pod [flags]",
-		Short:        "evict the pod selected by name or labels",
+		Short:        "evict selected pods",
 		SilenceUsage: true,
 		RunE: func(c *cobra.Command, args []string) error {
 			if o.printVersion {
 				fmt.Println(Version)
 				os.Exit(0)
 			}
 
-			if err := o.Complete(c, args); err != nil {
+			if err := o.Validate(args); err != nil {
```

Comment:
I believe `Complete` needs to be run before `Validate`? IIRC, this is the pattern in kubectl implementations, where `Complete` function is used for populating info from the flags etc, and then `Validate` validates if that info is sufficient or not.

e.g. https://github.com/kubernetes/kubectl/blob/2e31adb070639fef809fe111475be66dfb35beec/pkg/cmd/delete/delete.go#L156

Reply:
kk, restructed so that Complete sets config and Validate validates config
just did not like the previous mix-and-match